contracts/Datastore.sol

```diff
@@ -321,29 +321,33 @@ contract Datastore is AragonApp {
     }
 
     /**
-     * @notice Sets IPFS as the storage provider for the datastore.
+     * @notice Sets the storage provider for the datastore
      * @dev Since switching between storage providers is not supported,
      * the method can only be called if storage isn't set or already IPFS.
      * Also sets AES as the encryption provider.
+     * @param _storageProvider Storage provider
      * @param _host Host
      * @param _port Port
      * @param _protocol HTTP protocol
      * @param _name Name of the AES encryption algorithm
      * @param _length Length of the encryption key
      */
-    function setSettings(string _host, uint16 _port, string _protocol, string _name, uint256 _length) public {
-        require(settings.storageProvider == StorageProvider.None || settings.storageProvider == StorageProvider.Ipfs);
+    function setSettings(StorageProvider _storageProvider, string _host, uint16 _port, string _protocol, string _name, uint256 _length) public {
+        require(settings.storageProvider == StorageProvider.None || settings.storageProvider == StorageProvider.Ipfs || settings.storageProvider == StorageProvider.Swarm);
```

Comment:
I think the goal of this `require` is to prevent users to change the storageProvider if it's already set. Perhaps it would be better if we check if `settings.storageProvider` is `None` or if `settings.storageProvider` is equal to what's already set in the smart contract. What do you think?
```suggestion
        require(settings.storageProvider == StorageProvider.None || settings.storageProvider == _storageProvider);
```

Reply:
Yep! Made the change :)